File: javascript/resistor-color/resistor-color.js
export const colorCode = code => {
  let i;
  for (i = 0; i < COLORS.length; i++) {
    if (COLORS[i] === code) {
      break;
    }
  }
  return i;
};

export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
];

File: ruby/resistor-color/resistor_color.rb
# frozen_string_literal: true

class ResistorColor
  COLORS = %w[black brown red orange yellow green blue violet grey white].freeze

  def self.color_code(color)
    index = 0
    done = false

    until done
      if color == COLORS[index]
        done = true
        break
      end

      index += 1
    end

    index
  end
end
